Formula/p/python@3.14.rb

```diff
@@ -0,0 +1,504 @@
+class PythonAT314 < Formula
+  desc "Interpreted, interactive, object-oriented programming language"
+  homepage "https://www.python.org/"
+  url "https://www.python.org/ftp/python/3.14.0/Python-3.14.0.tgz"
+  sha256 "88d2da4eed42fa9a5f42ff58a8bc8988881bd6c547e297e46682c2687638a851"
+  license "Python-2.0"
+
+  livecheck do
+    url "https://www.python.org/ftp/python/"
+    regex(%r{href=.*?v?(3\.14(?:\.\d+)*)/?["' >]}i)
+  end
+
+  depends_on "pkgconf" => :build
+  depends_on "mpdecimal"
+  depends_on "openssl@3"
+  depends_on "sqlite"
+  depends_on "xz"
+
```

Comment:
Hi! Wanted to make sure that the new [`compression.zstd` module](https://docs.python.org/3/library/compression.zstd.html) is enabled for 3.14. I am not super familiar with Homebrew, but I think this is all that should be required to pull in the library that autoconf will pick up?
```suggestion
  depends_on "zstd"
```

Reply:
@branchvincent zstd support is mentioned as a "significant improvements in the standard library" in the [docs](https://docs.python.org/3/whatsnew/3.14.html) and [release notes](https://www.python.org/downloads/release/python-3140/). The official macOS Python installed supports it. I think it makes sense to include this dependency